services/venice-controller/src/main/java/com/linkedin/venice/controller/Admin.java

```diff
@@ -690,6 +692,17 @@ void updateRoutersClusterConfig(
    */
   boolean truncateKafkaTopic(String topicName);
 
+  /**
+   * Truncate a Kafka topic by setting its retention time to the input value.
+   * @param topicName the name of the topic to truncate.
+   * @param retentionTimeInMs the retention time in milliseconds to set for the topic.
+   * @return true if truncating this topic successfully.
+   *        false otherwise.
+   */
+  default boolean truncateKafkaTopic(String topicName, Optional<Long> retentionTimeInMs) {
```

Comment:
Nit,
Do we need to use `Optional` param for `retentionTimeInMs` considering there is already a function is not using this param?

Reply:
Do we need to offer a default impl?

Reply:
I think it makes sense and  we can simplify this.